apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -549,13 +549,21 @@ export default class CoreLibrary {
   }
 
   runSpriteCreatedEvents(newSprite) {
-    const matchingInputEvents = this.inputEvents.filter(
-      inputEvent =>
-        inputEvent.type === 'whenSpriteCreated' &&
-        (inputEvent.args.name === newSprite.name ||
-          inputEvent.args.costume === newSprite.getAnimationLabel() ||
-          inputEvent.args.costume === 'all')
-    );
+    const matchingInputEvents = this.inputEvents.filter(inputEvent => {
+      if (inputEvent.type === 'whenSpriteCreated') {
+        const inputEventName = inputEvent.args.name;
+        const inputEventCostume = inputEvent.args.costume;
+        if (inputEventName !== undefined && inputEventName === newSprite.name) {
```

Comment:
Feels a little weird to me that both the if and else if here return true?

Could you just replace the problematic condition in the existing code with your updated condition, or did that just feel too confusing?

Reply:
Sure. I thought the condition was getting kinda long so changed it to this way, but can just go with the longer condition which looks like this. Either way works.
```
    const matchingInputEvents = this.inputEvents.filter(
      inputEvent =>
        inputEvent.type === 'whenSpriteCreated' &&
        ((inputEvent.args.name !== undefined &&
          inputEvent.args.name === newSprite.name) ||
          inputEvent.args.costume === newSprite.getAnimationLabel() ||
          inputEvent.args.costume === 'all')
    );
```
Lmk what you think - thanks!

Reply:
Yeah agree kind of long, but if we have to pick between the two I think this is clearer than the other version